### DISCUSSION TITLE
Poetry seems install with different hash value from the one in lock file

### DISCUSSION THREAD

**Discussion by mockingjet** [python-poetry/poetry#7020] [Q&A] [answered]
Installation works well with docker python3.9 image (on windows) on my dev server.

But when I install with the same docker image (on ubuntu) on the production server. Poetry emits error: **Hash for tensorflow (2.6.0) from archive tensorflow-2.6.0-cp39-cp39-manylinux2010_x86_64.whl not found in known hashes (was: sha256:013529b6d7eeaf37da641b7233606f7dc0cbab71de7808c1dc7278e0eb2ded5b)**
![image](https://user-images.githubusercontent.com/32839503/201506346-9d828686-b8ba-4b13-9550-e9f713470bab.png)

I doubt that poetry points to wrong hash. Here is lock file hash value.
![image](https://user-images.githubusercontent.com/32839503/201506411-3351a9b1-334f-40dd-81bd-0653b1a5f39c.png)

Is this a bug?

**UPDATE**: For anyone who might have the same issue, the root cause was **OOM**.

**Comment by neersighted** [comment, marked as answer]
The [correct hash](https://pypi.org/project/tensorflow/2.6.0/#copy-hash-modal-1609b9ab-c6c2-404a-a186-dcc6dd5578c3) is recorded in the lock file -- I suspect you have a corrupt download. Try blowing away `$(poetry config cache-dir)/artifacts`.

**Reply by mockingjet** [reply]
Could you give me further detailed instructions?
I don't get what you mean by *Try blowing away $(poetry config cache-dir)/artifacts.*

**Reply by neersighted** [reply]
Delete the entire directory at that path. That is to say, the directory at the path output by `poetry config cache-dir` plus `/artifacts`.

**Reply by mockingjet** [reply]
I removed `artifacts`, and it still emit the error.
![image](https://user-images.githubusercontent.com/32839503/201567086-25b396bd-d0f1-4e3d-9d1c-f264374ff345.png)

I can't imagine how I got  corrupt download, since I did the same as what I did on windows ( the same docker-compose ).

Even I remove lock file and reinstall, still get killed for unknown reason
![image](https://user-images.githubusercontent.com/32839503/201569892-41620390-3092-45f4-8369-8fb52d8f7e9f.png)

**Reply by neersighted** [reply]
I'm unable to reproduce with the following -- seems to be a local issue. I would make sure your cache really is clear (or try a `poetry install -vvv --no-cache`)...

```
$ docker run --rm -it --entrypoint /bin/bash python:3.9
# pip install poetry==1.2.2
# poetry new foo && cd foo
# poetry add tensorflow==2.6.0
```

**Reply by mockingjet** [reply]
Do you use ubuntu? I tried your scripts, and still got `killed`. I should change to a new server, doubting that it could be memory issue. ( I am using t2.mirco now. )

**Reply by neersighted** [reply]
It does not matter what my host OS is (indeed, I cannot reproduce your issue on any one of macOS, Linux, or Windows); those are steps performed in a container (using the `python:3.9` tag, which currently points at `python@sha256:475fe86ebf1da48ea27009a8f7d7e96231af4142de918a68010d48d0abb9c9c5`). The `$` is on the host system, the `#` is inside the container. `killed` is not the same as a hash mismatch (and it looks like I replied before your edit) -- that's likely the OOM killer; you should check your syslog. I suspect that you do not have enough memory to extract the Tensorflow archive.

**Reply by mockingjet** [reply]
After I change to a new server (t2.small) and it works. I inspected the server log and it was really caused by OOM. And docker restart mechanism will run `poetry install` again, that's where invalid hash happens, though I don't know what's the connection... But thank you! You help me a lot.
